modules/sdk/src/main/scala/org/tessellation/sdk/http/p2p/clients/L0GlobalSnapshotClient.scala

```diff
@@ -4,82 +4,40 @@ import cats.effect.Async
-    new L0GlobalSnapshotClient[F] {
-
-      def getLatestOrdinal: PeerResponse[F, SnapshotOrdinal] = {
-        import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
-
-        implicit val decoder: Decoder[SnapshotOrdinal] = deriveMagnoliaDecoder[SnapshotOrdinal]
-
-        PeerResponse[F, SnapshotOrdinal]("global-snapshots/latest/ordinal")(client)
-      }
-
-      def getLatestMetadata: PeerResponse[F, SnapshotMetadata] = {
-        import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
-
-        PeerResponse[F, SnapshotMetadata]("global-snapshots/latest/metadata")(client)
-      }
-
-      def getLatest: PeerResponse[F, (Signed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] = {
-        import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
-
-        PeerResponse[F, (Signed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)]("global-snapshots/latest/combined")(client)
-      }
-
-      def get(ordinal: SnapshotOrdinal): PeerResponse[F, Signed[GlobalIncrementalSnapshot]] = {
-        import org.tessellation.ext.codecs.BinaryCodec.decoder
-
-        PeerResponse[F, Signed[GlobalIncrementalSnapshot]](s"global-snapshots/${ordinal.value.value}")(client)
-      }
-
-      def getHash(ordinal: SnapshotOrdinal): PeerResponse[F, Option[Hash]] = {
-        import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
-
-        PeerResponse(s"global-snapshots/${ordinal.value.value}/hash", GET)(client) { (req, client) =>
-          client.expectOption[Hash](req)
-        }
-      }
-
-      def get(hash: Hash): PeerResponse[F, Signed[GlobalIncrementalSnapshot]] = {
-        import org.tessellation.ext.codecs.BinaryCodec.decoder
-
-        PeerResponse[F, Signed[GlobalIncrementalSnapshot]](s"global-snapshots/$hash")(client)
-      }
-
-      def getFull(ordinal: SnapshotOrdinal): PeerResponse[F, Signed[GlobalSnapshot]] = {
-        import org.tessellation.ext.codecs.BinaryCodec.decoder
-
-        PeerResponse[F, Signed[GlobalSnapshot]]((uri: Uri) =>
-          uri.addPath(s"global-snapshots/${ordinal.value.value}").withQueryParam("full")
-        )(client)
-      }
+  def make[F[_]: Async: SecurityProvider: KryoSerializer](
+    client: Client[F],
+    maybeSession: Option[Session[F]] = None
+  ): L0GlobalSnapshotClient[F] =
+    new L0GlobalSnapshotClientImpl[F](client, maybeSession)
+
+  private class L0GlobalSnapshotClientImpl[F[_]: Async: SecurityProvider: KryoSerializer](
+    client: Client[F],
+    maybeSession: Option[Session[F]]
+  ) extends SnapshotClient[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo](client, maybeSession)
+      with L0GlobalSnapshotClient[F] {
+
+    val urlPrefix = "global-snapshots"
+
+    def getFull(ordinal: SnapshotOrdinal): PeerResponse[F, Signed[GlobalSnapshot]] = {
+      import org.tessellation.ext.codecs.BinaryCodec.decoder
+
+      PeerResponse[F, Signed[GlobalSnapshot]]((uri: Uri) => uri.addPath(s"$urlPrefix/${ordinal.value.value}").withQueryParam("full"))(
+        client
+      )
     }
+  }
```

Comment:
one tip. you can use an anonymous class, instead of `Impl` suffix.

```
def make[F[_]: ...](...): L0GlobalSnapshotClient[F] = new SnapshotClient[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo](client, maybeSession) extends L0GlobalSnapshotClient {
  // implement methods there
}
```
or some kind of that

Reply:
Do you have a working example? You can't do it that way because `SnapshotClient` is abstract - compiler won't let us create it.

Reply:
I think you can create an instance of an abstract class. For sure you can create an instance of a trait. We use that in almost all storages/services. I cannot send you the link now but I'm sure you can quickly find that in repo

Reply:
I couldn't get it to work that way but I removed the constructor args from the abstract class and did it that way.